lib/ddtrace/contrib/http/patcher.rb

```diff
@@ -85,12 +87,16 @@ def patched?
         def patch_http
           ::Net::HTTP.class_eval do
             alias_method :initialize_without_datadog, :initialize
+
             Datadog::Patcher.without_warnings do
-              remove_method :initialize
+              remove_method :initialize if private_instance_methods(false).include?(:initialize)
             end
 
             def initialize(*args)
-              pin = Datadog::Pin.new(SERVICE, app: APP, app_type: Datadog::Ext::AppTypes::WEB)
+              service = Datadog.configuration[:http][:service_name]
+              tracer = Datadog.configuration[:http][:tracer]
+
+              pin = Datadog::Pin.new(service, app: APP, app_type: Datadog::Ext::AppTypes::WEB, tracer: tracer)
```

Comment:
So the following is a really nuanced thing, related to [a bug I stumbled upon in our Sequel integration](https://github.com/DataDog/dd-trace-rb/pull/408/files#diff-1291bc9b550762dc47c55c5ed0f08730L16)...

It's possible that an `Net::HTTP` object is initialized before Datadog is patched/configured, in which case, the object in question will not trace properly. To avoid this scenario, I would suggest refactoring this pin initialization into its own `def datadog_pin` function, which lazily initializes the pin.

The other benefit you'd get is that you can remove the `#initialize` patching altogether from this, making the patch less invasive.

Reply:
yup lazy initialization seems to work. Thanks for suggesting it, it made the code much cleaner